File: QytetetJava/src/modeloqytetet/Dado.java
/**
 * Practica de:
 * 
 * Francisco Beltrán Sánchez
 * Pablo Cordero Romero
 */
package modeloqytetet;
import java.util.Random;

public class Dado {
    private static final Dado instance = new Dado();
    private int valor;
    
    int tirar(){
        Random rnd = new Random();
        valor = rnd.nextInt(6)+1;
        
        return valor;
    }
    
    private Dado(){
        valor = 0;
    
    }
    
    public int getValor(){
        return valor;
    }
    
    public static Dado getInstance(){
        return instance;
    }
    
    @Override
    public String toString() {
        return "\n------\nTablero" + "\n\tvalor=" + valor ;
    }
}

File: QytetetRuby/lib/dado.rb
#encoding: utf-8
# Francisco Beltrán Sánchez
# Pablo Cordero Romero
require "singleton"

module ModeloQytetet
  class Dado
    include Singleton
    attr_reader :valor
    
    def initialize
       @valor
    end
    
    def tirar
      @valor = rand(1..6)
      @valor
    end
    
    def to_s
      "Dado \n Valor #{@valor}"
    end
    
  end
end
